packages/synapse-react-client/src/components/EntityTreeTable/components/EntityTreeTableView.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import { Table, Row, flexRender } from '@tanstack/react-table'
+import { EntityBundleRow } from '../EntityTreeTable'
+import { AutoLoadMore } from './AutoLoadMore'
+import { useEntityTreeTableContext } from './EntityTreeTableContext'
+import StyledTanStackTable from '../../TanStackTable/StyledTanStackTable'
+import { TrProps } from '../../TanStackTable/types'
+
+type EntityTreeTableViewProps = {
+  table: Table<EntityBundleRow>
+}
+
+// Custom table row component to handle load-more rows
+const EntityTreeTableRow: React.FC<
+  TrProps<EntityBundleRow, Row<EntityBundleRow>>
+> = ({ row, tableRow, ...rest }) => {
+  const { loadingIds, loadMoreChildren } = useEntityTreeTableContext()
+
+  if (!tableRow) return null
+
+  const original = tableRow.original
+  if (original.isLoadMore) {
+    const colCount = tableRow.getAllCells().length
+    const isLoading = loadingIds.has(original.parentId!)
+
+    return (
+      <tr {...rest}>
+        <td colSpan={colCount}>
+          <AutoLoadMore
+            depth={original.depth}
+            isLoading={isLoading}
+            pageToken={original.pageToken}
+            onLoadMore={() => {
+              loadMoreChildren(original.parentId!, original.pageToken)
+            }}
+          />
+        </td>
+      </tr>
+    )
+  }
+
+  // For normal rows, render cells normally
+  return (
+    <tr {...rest}>
+      {tableRow.getVisibleCells().map(cell => (
+        <td
```

Comment:
Is it possible to use the default `TableCellRenderer` here? To avoid duplicate code/having to change this in multiple places if we try to make an update to the shared table components